src/components/Panel/index.ts

```diff
@@ -12,50 +12,101 @@ const CLASS_PANEL_HORIZONTAL = CLASS_PANEL + '-horizontal';
 const CLASS_PANEL_SORTABLE_ICON = CLASS_PANEL + '-sortable-icon';
 const CLASS_PANEL_REMOVE = CLASS_PANEL + '-remove';
 
-// TODO: document panelType
-
-/**
- * @event
- * @name Panel#collapse
- * @description Fired when the panel gets collapsed
- */
-
-/**
- * @event
- * @name Panel#expand
- * @description Fired when the panel gets expanded
- */
+namespace Panel {
+    export interface Args extends Container.Args {
+    /**
+     * Sets whether the Element is collapsible.
+     */
+    collapsible?: boolean,
+    /**
+     * Sets whether the Element should be collapsed.
+     */
+    collapsed?: number,
+    /**
+     * Sets whether the panel can be reordered
+     */
+    sortable?: boolean,
+    /**
+     * Sets whether the panel collapses horizontally - this would be the case for side panels. Defaults to false.
+     */
+    collapseHorizontally?: boolean,
+    /**
+     * Sets whether the panel can be removed
+     */
+    removable?: boolean,
+    /**
+     * The height of the header in pixels. Defaults to 32.
+     */
+    headerSize?: number,
+    /**
+     * The header text of the panel. Defaults to the empty string.
+     */
+    headerText?: string,
+    /**
+     * Sets the panel type.
```

Comment:
Enumerate the accepted values. Is it also worth tightening the type to string literals?